src/BloomExe/Book/BookStorage.cs

```diff
@@ -393,57 +398,42 @@ internal static List<string> GetImagePathsRelativeToBook(XmlElement element)
 		/// in the dom, and remove any unreferenced ones.
 		/// </summary>
 		public void CleanupUnusedAudioFiles()
 		{
-
 			if (IsStaticContent(_folderPath))
 				return;
+
 			//Collect up all the audio files in our book's audio directory
 			var audioFolderPath = AudioProcessor.GetAudioFolderPath(_folderPath);
-			var audioFilesToDeleteIfNotUsed = new List<string>();
+			HashSet<string> audioFilesToDeleteIfNotUsed;    // Should be a HashSet or something we can delete a specific item from repeatedly
 
 			if (Directory.Exists(audioFolderPath))
 			{
-				foreach (var path in Directory.EnumerateFiles(audioFolderPath).Where(
-					s => AudioExtensions.Contains(Path.GetExtension(s).ToLowerInvariant())))
-				{
-					audioFilesToDeleteIfNotUsed.Add(Path.GetFileName(GetNormalizedPathForOS(path)));
-				}
+				// Review: do we only want to delete files in this directory if they have certain file extensions?
+				audioFilesToDeleteIfNotUsed = new HashSet<string>(Directory.EnumerateFiles(audioFolderPath).Where(AudioProcessor.IsAudioFile)
+					.Select(path => Path.GetFileName(GetNormalizedPathForOS(path))));
 			}
-
-			//Remove from that list each audio file actually in use
-			var element = Dom.RawDom.DocumentElement;
-			var usedAudioPaths = GetAudioPathsRelativeToBook(element);
-
-			//also, ensure that nothing to be removed is actually used in a datadiv as audio-sentence or backgroundaudio.
-			//This prevents us from deleting, for example, cover page audios if this is called before the front-matter
-			//has been applied to the document.
-			usedAudioPaths.AddRange(from XmlElement dataDivAudio in Dom.RawDom.SelectNodes("//div[@id='bloomDataDiv']//div[contains(text(),'audio-sentence')"+
-							" or contains(text(),'data-backgroundaudio')]")
-							  select UrlPathString.CreateFromUrlEncodedString(dataDivAudio.InnerText.Trim()).PathOnly.NotEncoded);
-			foreach (var fileName in usedAudioPaths)
+			else
 			{
-				if (Path.GetExtension(fileName).Length > 0)
-				{
-					if (AudioExtensions.Contains(Path.GetExtension(fileName).ToLowerInvariant()))
-					{
-						audioFilesToDeleteIfNotUsed.Remove(GetNormalizedPathForOS(fileName));  //This call just returns false if not found, which is fine.
-					}
-				}
-				foreach (var ext in AudioExtensions)
-				{
-					var tempfileName = Path.ChangeExtension(fileName, ext);
-					audioFilesToDeleteIfNotUsed.Remove(GetNormalizedPathForOS(tempfileName));
-				}
+				audioFilesToDeleteIfNotUsed = new HashSet<string>();
 			}
+
+			//Look up which files could actually be used by the book
+			var usedAudioFileNames = new HashSet<string>();
+
+			var backgroundMusicFileNames = GetBackgroundMusicFileNamesReferencedInBook();
+			usedAudioFileNames.AddRange(backgroundMusicFileNames);
+
+			var narrationAudioFileNames = GetNarrationAudioFileNamesReferencedInBook(true);
+			usedAudioFileNames.AddRange(narrationAudioFileNames);
+
+			foreach (var fileName in usedAudioFileNames)
```

Comment:
now that we refactored out the part that figured out which file names we cared about, we can simplify this and don't need to manually intersect the lists anymore.  Could replace the loop with:

audioFilesToDeleteIfNotUsed.ExceptWith(usedAudioFileNames)